Exploration-of-the-Oppia-stack.md

```diff
@@ -0,0 +1,431 @@
+## Table of Contents
+
+* [Support editing](#support-editing)
+  * [Frontend changes to support editing](#frontend-changes-to-support-editing)
+  * [Backend changes to support editing](#backend-changes-to-support-editing)
+* [Support viewing](#support-viewing)
+  * [Backend changes to support viewing](#backend-changes-to-support-viewing)
+  * [Frontend changes to support viewing](#frontend-changes-to-support-viewing)
+
+To better understand how Oppia works, let's work through a simple full-stack change--adding a field to a user's preferences. We'll discuss how to update the frontend code to display the new field and how to make the backend store this new field in the database.
```

Comment:
This still sounds like a wiki page explaining how to do full-stack changes! I'm worried that the contributor will start using this as a wiki page for making full stack changes! Should we explain this using an existing field in the codebase?

Reply:
I think the value of this page is that it's an active way to learn how Oppia works. Readers get to change the code and see the effects ("learn by doing") instead of just reading. If we used an existing field, then this would become just an exercise in reading code and the wiki page, which I think would be much less instructive.

Regardless, I appreciate your feedback that this still sounds like instructions for how to do a full-stack change. I'm stuck though on how to fix this problem because to my eye it seems clear that the page is an exploration of how Oppia works, not a guide to writing production code. Do you have any suggestions for how to make this clearer?